Add the NIOS RECORD PTR Module

clementtrebuchet opened the pull request:
##### SUMMARY
<!--- Describe the change, including rationale and design decisions -->

The PTR module would allow the recording of an individual reverse DNS record

<!--- If you are fixing an existing issue, please include "Fixes #nnn" in your
commit message and your description; but you should still explain what
the change does.-->

Fixes #41090

##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - New Module Pull Request

##### COMPONENT NAME
<!--- Name of the module, plugin, module or task -->
nios_ptr_record

##### ANSIBLE VERSION
<!--- Paste verbatim output from "ansible --version" between quotes below -->
```
(venv) [clement@clemo ansible]$ ansible --version
ansible 2.7.0.dev0 (devel_nios_ptr_record 7cd0f51541) last updated 2018/06/25 14:01:36 (GMT +200)
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/clement/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/clement/Documents/PROJECTS/ansible_env_setup/ansible/lib/ansible
  executable location = /home/clement/Documents/PROJECTS/ansible_env_setup/ansible/bin/ansible
  python version = 3.6.5 (default, Mar 29 2018, 18:20:46) [GCC 8.0.1 20180317 (Red Hat 8.0.1-0.19)]

```


##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here.
For bugs that don't have a linked bug report, a step-by-step reproduction
of the problem is helpful. -->

![image](https://user-images.githubusercontent.com/8559910/41849651-f28d2378-7881-11e8-95e4-c5c38fc818d5.png)

<!--- Paste verbatim command output below, e.g. before and after your change -->
```

```

clementtrebuchet committed:
Add the NIOS RECORD PTR Module

clementtrebuchet committed:
FIX E241,E231, E105

clementtrebuchet committed:
FIX module should not be executable

clementtrebuchet committed:
FIX module RWX

clementtrebuchet committed:
Remove change in nios_srv_record

clementtrebuchet committed:
Remove change in nios_srv_record

clementtrebuchet committed:
FIX test_nios_ptr_record_update_comment

justjais commented:
@clementtrebuchet Thank you for your contribution, can you please write Integration and target tests for this module file.

clementtrebuchet commented:
OK I will produce it ASAP.

Thanks

clementtrebuchet committed:
ADD: Integration and target tests for the NIOS PTR RECORD module.

clementtrebuchet committed:
FIX module name in integrations tests

clementtrebuchet committed:
Update integration

clementtrebuchet committed:
Refactor nios_ptr_record module

clementtrebuchet committed:
Refactor the nios_ptr_record_idempotence

clementtrebuchet committed:
FIX name

clementtrebuchet committed:
Smoketests for the NIOS modules does not take care of the PTR:RECORD object

mattclay commented:
CI failure in integration tests: https://app.shippable.com/github/ansible/ansible/runs/72431/66/tests

clementtrebuchet commented:
Hello @justjais 
It seems that your `Flask Infoblox Server` from the test container: 
https://github.com/ansible/nios-test-container/blob/master/flaskapp.py  
*does not handle the record:ptr object*

```python
[]
127.0.0.1 - - [03/Jul/2018 20:13:24] "GET /wapi/v2.1/record%3Aptr?ipv4addr=192.168.10.1&ptrdname=host.ansible.com&_return_fields=name%2Cview%2Cipv4addr%2Cptrdname%2Cttl%2Cextattrs%2Ccomment&_max_results=1000&_proxy_search=GM HTTP/1.1" 200 -
# METHOD: POST VIEWTYPE: record:ptr
# REQARGS ...
{}
# REQJSON ...
{'ipv4addr': '192.168.10.1', 'ptrdname': 'host.ansible.com'}
# CREATE VIEWTYPE: record:ptr
########### VIEWTYPE: record:ptr
127.0.0.1 - - [03/Jul/2018 20:13:25] "POST /wapi/v2.1/record%3Aptr HTTP/1.1" 500 -
Traceback (most recent call last):
  File "/home/clement/Documents/PROJECTS/ansible_env_setup/ansible/venv/lib/python3.6/site-packages/flask/app.py", line 2309, in __call__
    return self.wsgi_app(environ, start_response)
  File "/home/clement/Documents/PROJECTS/ansible_env_setup/ansible/venv/lib/python3.6/site-packages/flask/app.py", line 2295, in wsgi_app
    response = self.handle_exception(e)
  File "/home/clement/Documents/PROJECTS/ansible_env_setup/ansible/venv/lib/python3.6/site-packages/flask/app.py", line 1741, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/home/clement/Documents/PROJECTS/ansible_env_setup/ansible/venv/lib/python3.6/site-packages/flask/_compat.py", line 35, in reraise
    raise value
  File "/home/clement/Documents/PROJECTS/ansible_env_setup/ansible/venv/lib/python3.6/site-packages/flask/app.py", line 2292, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/clement/Documents/PROJECTS/ansible_env_setup/ansible/venv/lib/python3.6/site-packages/flask/app.py", line 1815, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/clement/Documents/PROJECTS/ansible_env_setup/ansible/venv/lib/python3.6/site-packages/flask/app.py", line 1718, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/home/clement/Documents/PROJECTS/ansible_env_setup/ansible/venv/lib/python3.6/site-packages/flask/_compat.py", line 35, in reraise
    raise value
  File "/home/clement/Documents/PROJECTS/ansible_env_setup/ansible/venv/lib/python3.6/site-packages/flask/app.py", line 1813, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/clement/Documents/PROJECTS/ansible_env_setup/ansible/venv/lib/python3.6/site-packages/flask/app.py", line 1799, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/clement/Documents/PROJECTS/playbooks/nios_ansible_ptr_module/nios-test-container/flaskapp.py", line 407, in v21_base
    view = DATA.create_view(request.get_json(), viewtype=viewtype)
  File "/home/clement/Documents/PROJECTS/playbooks/nios_ansible_ptr_module/nios-test-container/flaskapp.py", line 240, in create_view
    self.views[viewtype].append(view)
KeyError: 'record:ptr'
```
Adding the record:ptr in views did the trick.

```bash
diff --git a/flaskapp.py b/flaskapp.py
index 315eb50..737c34b 100644
--- a/flaskapp.py
+++ b/flaskapp.py
@@ -210,7 +210,8 @@ class DataModel(object):
             'ipv6network': [NetworkView(uid='ZG5zLm5ldHdvcmskZmU4MDo6LzY0LzA', isdefault=True, name='default', network='fe80::/64')],
             'zone_auth': [],
             'view': [NetworkView(isdefault=True, name='default', viewtype='view')],
-            'record:host': []
+            'record:host': [],
+            'record:ptr': [],
         }
         # ZG5zLm5ldHdvcmtfdmlldyQw == dns.network_view$0
         # ZG5zLm5ldHdvcmskZmU4MDo6LzY0LzA == dns.network$fe80::/64
@@ -477,4 +478,4 @@ def v21_abstractview_ref(viewtype, refid=None, subname=None, refpath=None, subsu
 
 
 if __name__ == "__main__":
-    app.run(host='0.0.0.0', port=443, ssl_context='adhoc')
+    app.run(host='0.0.0.0', port=443, ssl_context='adhoc', debug=True)

```
*The Integration and target tests pass*

```bash
(venv) [clement@clemo nios_ansible_ptr_module]$ ansible-playbook -i inventory  nios_ptr_record_test_playbook.yml  --e="ansible_python_interpreter=$(which python)"

PLAY [DNS] *********************************************************************************************************************************************************************

TASK [nios_ptr : create an ipv4 ptr record ptrdname=host.ansible.com, ipv4=192.168.10.1, state=present, provider={{ nios_provider }}] ******************************************
Tuesday 03 July 2018  21:28:45 +0200 (0:00:00.234)       0:00:00.234 ********** 
changed: [127.0.0.1]

TASK [nios_ptr : add a comment to an existing ipv4 ptr record ptrdname=host.ansible.com, ipv4=192.168.10.1, comment=this is a test comment, state=present, provider={{ nios_provider }}] ***
Tuesday 03 July 2018  21:28:46 +0200 (0:00:01.006)       0:00:01.241 ********** 
changed: [127.0.0.1]

TASK [nios_ptr : remove a ptr record from the system ptrdname=host.ansible.com, ipv4=192.168.10.1, state=absent, provider={{ nios_provider }}] *********************************
Tuesday 03 July 2018  21:28:47 +0200 (0:00:00.845)       0:00:02.086 ********** 
changed: [127.0.0.1]

TASK [nios_ptr : assert that=['ipv4_ptr_create1.changed', 'ipv4_ptr_update1.changed', 'ipv4_ptr_delete1.changed']] *************************************************************
Tuesday 03 July 2018  21:28:47 +0200 (0:00:00.838)       0:00:02.924 ********** 
ok: [127.0.0.1] => {
    "changed": false,
    "msg": "All assertions passed"
}

PLAY RECAP *********************************************************************************************************************************************************************
127.0.0.1                  : ok=4    changed=3    unreachable=0    failed=0   

Tuesday 03 July 2018  21:28:47 +0200 (0:00:00.057)       0:00:02.982 ********** 
=============================================================================== 
nios_ptr : create an ipv4 ptr record ------------------------------------------------------------------------------------------------------------------------------------ 1.01s
nios_ptr : add a comment to an existing ipv4 ptr record ----------------------------------------------------------------------------------------------------------------- 0.85s
nios_ptr : remove a ptr record from the system -------------------------------------------------------------------------------------------------------------------------- 0.84s
nios_ptr : assert ------------------------------------------------------------------------------------------------------------------------------------------------------- 0.06s
Playbook run took 0 days, 0 hours, 0 minutes, 2 seconds

```
*As well in the fake infoblox server*

```bash
# METHOD: GET VIEWTYPE: record:ptr
# REQARGS ...
{'_max_results': '1000',
 '_return_fields': 'name,view,ipv4addr,ptrdname,ttl,extattrs,comment',
 'ipv4addr': '192.168.10.1',
 'ptrdname': 'host.ansible.com'}
# REQJSON ...
{}
# FETCHED VIEW ...
# collecting all in record:ptr
[]
127.0.0.1 - - [03/Jul/2018 21:28:46] "GET /wapi/v2.1/record%3Aptr?ipv4addr=192.168.10.1&ptrdname=host.ansible.com&_return_fields=name%2Cview%2Cipv4addr%2Cptrdname%2Cttl%2Cextattrs%2Ccomment&_max_results=1000 HTTP/1.1" 200 -
# METHOD: GET VIEWTYPE: record:ptr
# REQARGS ...
{'_max_results': '1000',
 '_proxy_search': 'GM',
 '_return_fields': 'name,view,ipv4addr,ptrdname,ttl,extattrs,comment',
 'ipv4addr': '192.168.10.1',
 'ptrdname': 'host.ansible.com'}
# REQJSON ...
{}
# FETCHED VIEW ...
# collecting all in record:ptr
[]
127.0.0.1 - - [03/Jul/2018 21:28:46] "GET /wapi/v2.1/record%3Aptr?ipv4addr=192.168.10.1&ptrdname=host.ansible.com&_return_fields=name%2Cview%2Cipv4addr%2Cptrdname%2Cttl%2Cextattrs%2Ccomment&_max_results=1000&_proxy_search=GM HTTP/1.1" 200 -
# METHOD: POST VIEWTYPE: record:ptr
# REQARGS ...
{}
# REQJSON ...
{'ipv4addr': '192.168.10.1', 'ptrdname': 'host.ansible.com'}
# CREATE VIEWTYPE: record:ptr
########### VIEWTYPE: record:ptr
# CREATED VIEW ...
{'_ref': 'record:ptr/ZG5zLnJlY29yZDpwdHIkTm9uZQ==:/false',
 'comment': None,
 'extattrs': {},
 'fqdn': None,
 'ipv4addrs': [],
 'ipv6addrs': [],
 'is_default': False,
 'name': '',
 'network': None,
 'network_view': 'default',
 'options': [],
 'uid': 'ZG5zLnJlY29yZDpwdHIkTm9uZQ==',
 'view': None,
 'viewtype': 'record:ptr'}
127.0.0.1 - - [03/Jul/2018 21:28:46] "POST /wapi/v2.1/record%3Aptr HTTP/1.1" 201 -
# METHOD: GET VIEWTYPE: record:ptr
# REQARGS ...
{'_max_results': '1000',
 '_return_fields': 'name,view,ipv4addr,ptrdname,ttl,extattrs,comment',
 'ipv4addr': '192.168.10.1',
 'ptrdname': 'host.ansible.com'}
# REQJSON ...
{}
# FETCHED VIEW ...
# collecting all in record:ptr
[{'_ref': 'record:ptr/ZG5zLnJlY29yZDpwdHIkTm9uZQ==:/false',
  'comment': None,
  'extattrs': {},
  'fqdn': None,
  'ipv4addrs': [],
  'ipv6addrs': [],
  'is_default': False,
  'name': '',
  'network': None,
  'network_view': 'default',
  'options': [],
  'uid': 'ZG5zLnJlY29yZDpwdHIkTm9uZQ==',
  'view': None,
  'viewtype': 'record:ptr'}]
127.0.0.1 - - [03/Jul/2018 21:28:46] "GET /wapi/v2.1/record%3Aptr?ipv4addr=192.168.10.1&ptrdname=host.ansible.com&_return_fields=name%2Cview%2Cipv4addr%2Cptrdname%2Cttl%2Cextattrs%2Ccomment&_max_results=1000 HTTP/1.1" 200 -
# REFPATH: ZG5zLnJlY29yZDpwdHIkTm9uZQ==:/false
# METHOD: PUT VIEWTYPE: record:ptr UID: false _REFID: record:ptr/ZG5zLnJlY29yZDpwdHIkTm9uZQ==:/false
# REQARGS ...
{}
# REQJSON ...
{'comment': 'this is a test comment',
 'ipv4addr': '192.168.10.1',
 'ptrdname': 'host.ansible.com'}
# UPDATING REFID record:ptr/ZG5zLnJlY29yZDpwdHIkTm9uZQ==:/false
# network/ZG5zLm5ldHdvcmtfdmlldyQw:1.0.0.0/24/default != record:ptr/ZG5zLnJlY29yZDpwdHIkTm9uZQ==:/false
# network/ZG5zLm5ldHdvcmtfdmlldyQw:1.0.0.0/24/default != record:ptr/ZG5zLnJlY29yZDpwdHIkTm9uZQ==:/false
# network/ZG5zLm5ldHdvcmskZmU4MDo6LzY0LzA:fe80::/64/default != record:ptr/ZG5zLnJlY29yZDpwdHIkTm9uZQ==:/false
# view/ZG5zLnZpZXckTm9uZQ==:default/true != record:ptr/ZG5zLnJlY29yZDpwdHIkTm9uZQ==:/false
# record:ptr/ZG5zLnJlY29yZDpwdHIkTm9uZQ==:/false == record:ptr/ZG5zLnJlY29yZDpwdHIkTm9uZQ==:/false
# pk: ipv4addr pv: 192.168.10.1
# pk: ptrdname pv: host.ansible.com
# pk: comment pv: this is a test comment
comment is "None" on record:ptr/ZG5zLnJlY29yZDpwdHIkTm9uZQ==:/false
setting comment to "this is a test comment" on record:ptr/ZG5zLnJlY29yZDpwdHIkTm9uZQ==:/false
# record:ptr/ZG5zLnJlY29yZDpwdHIkTm9uZQ==:/false updated? True
# viewk: record:ptr
# viewix: 0
# accessing found view
# return <__main__.NetworkView object at 0x7f9e59238080>
# MODIFIED VIEW [ZG5zLnJlY29yZDpwdHIkTm9uZQ==] ...
{'_ref': 'record:ptr/ZG5zLnJlY29yZDpwdHIkTm9uZQ==:/false',
 'comment': 'this is a test comment',
 'extattrs': {},
 'fqdn': None,
 'ipv4addrs': [],
 'ipv6addrs': [],
 'is_default': False,
 'name': '',
 'network': None,
 'network_view': 'default',
 'options': [],
 'uid': 'ZG5zLnJlY29yZDpwdHIkTm9uZQ==',
 'view': None,
 'viewtype': 'record:ptr'}
127.0.0.1 - - [03/Jul/2018 21:28:46] "PUT /wapi/v2.1/record%3Aptr/ZG5zLnJlY29yZDpwdHIkTm9uZQ%3D%3D%3A/false HTTP/1.1" 200 -
# METHOD: GET VIEWTYPE: record:ptr
# REQARGS ...
{'_max_results': '1000',
 '_return_fields': 'name,view,ipv4addr,ptrdname,ttl,extattrs,comment',
 'ipv4addr': '192.168.10.1',
 'ptrdname': 'host.ansible.com'}
# REQJSON ...
{}
# FETCHED VIEW ...
# collecting all in record:ptr
[{'_ref': 'record:ptr/ZG5zLnJlY29yZDpwdHIkTm9uZQ==:/false',
  'comment': 'this is a test comment',
  'extattrs': {},
  'fqdn': None,
  'ipv4addrs': [],
  'ipv6addrs': [],
  'is_default': False,
  'name': '',
  'network': None,
  'network_view': 'default',
  'options': [],
  'uid': 'ZG5zLnJlY29yZDpwdHIkTm9uZQ==',
  'view': None,
  'viewtype': 'record:ptr'}]
127.0.0.1 - - [03/Jul/2018 21:28:47] "GET /wapi/v2.1/record%3Aptr?ipv4addr=192.168.10.1&ptrdname=host.ansible.com&_return_fields=name%2Cview%2Cipv4addr%2Cptrdname%2Cttl%2Cextattrs%2Ccomment&_max_results=1000 HTTP/1.1" 200 -
# REFPATH: ZG5zLnJlY29yZDpwdHIkTm9uZQ==:/false
# METHOD: DELETE VIEWTYPE: record:ptr UID: false _REFID: record:ptr/ZG5zLnJlY29yZDpwdHIkTm9uZQ==:/false
# REQARGS ...
{}
# REQJSON ...
{}
# DELETE VIEW [record:ptr/ZG5zLnJlY29yZDpwdHIkTm9uZQ==:/false]
REMOVING record:ptr/ZG5zLnJlY29yZDpwdHIkTm9uZQ==:/false
# collecting all in record:ptr
[]
127.0.0.1 - - [03/Jul/2018 21:28:47] "DELETE /wapi/v2.1/record%3Aptr/ZG5zLnJlY29yZDpwdHIkTm9uZQ%3D%3D%3A/false HTTP/1.1" 200 -

```
So can you tell me how to handle this ?

mattclay commented:
@clementtrebuchet Please open a PR for nios-test-container with the necessary changes and reference this PR. Once those changes have been merged we can tag a new release of the container and re-run the tests on this PR.

clementtrebuchet committed:
REM the default for view

clementtrebuchet commented:
Hello @mattclay 

The PR https://github.com/ansible/nios-test-container/pull/1 is open.

Thanks,

clementtrebuchet committed:
Add finals test, after adding the PR (https://github.com/ansible/nios-test-container/pull/1) in order to handle PTR:RECORD

justjais commented:
@clementtrebuchet Thanks for writing the required integration and unit tests for the respective module. Also, I'll review and approve the changes you made in _flaskapp.py_.

Meanwhile can you please fix the shippable build error, so that we can approve and close the PR.

clementtrebuchet commented:
Hello @justjais

The only shippable build errors in this PR is related to the [Ansible NIOS test container](https://github.com/ansible/nios-test-container) who failed, because it does not handle the record:ptr object.
It is the purpose of the PR https://github.com/ansible/nios-test-container/pull/1, when this container will be updated, I will relaunch the CI and it will pass it. So relaunching the CI on this PR is useless until the PR https://github.com/ansible/nios-test-container/pull/1 is not integrated.

https://app.shippable.com/github/ansible/ansible/runs/72588/66/tests

![image](https://user-images.githubusercontent.com/8559910/42313421-02982060-8043-11e8-86e7-d3724a8221ca.png)

https://app.shippable.com/github/ansible/ansible/runs/72588/67/tests

![image](https://user-images.githubusercontent.com/8559910/42313467-281a018c-8043-11e8-9d43-307cc062dc43.png)

Best regards,

justjais commented on test/units/modules/net_tools/nios/test_nios_ptr_record.py:
```diff
@@ -0,0 +1,158 @@
+from .test_nios_module import TestNiosModule, load_fixture
+
+
+class TestNiosHostRecordModule(TestNiosModule):
```
Can you please change the class name to respective module name

clementtrebuchet replied:
Hello @justjais 

Ok done

justjais commented on test/units/modules/net_tools/nios/test_nios_ptr_record.py:
```diff
@@ -0,0 +1,158 @@
+
+    def setUp(self):
+
+        super(TestNiosHostRecordModule, self).setUp()
```
Same as above, please change the class name

clementtrebuchet replied:
Hello @justjais

Ok done

justjais commented on test/units/modules/net_tools/nios/test_nios_ptr_record.py:
```diff
@@ -0,0 +1,158 @@
+        res = wapi.run('testobject', test_spec)
+
+        self.assertTrue(res['changed'])
+        wapi.create_object.assert_called_once_with('testobject', {'ipv4addr': '10.36.241.14', 'ptrdname': 'ansible.test.com'})
```
can you please include test for ipv6addr as well, it will cover the complete workflow as for ipv4addr test is already written.

clementtrebuchet replied:
Hello @justjais 

Ok done.

justjais commented on lib/ansible/modules/net_tools/nios/nios_ptr_record.py:
```diff
@@ -0,0 +1,159 @@
+        del ib_spec['ipv4addr']
+
+    wapi = WapiModule(module)
+    result = wapi.run('record:ptr', ib_spec)
```
Can you please define nios specific **constant** i.e. `record:ptr` in `nios/api.py` folder, as from 2.6 onward we have nios constants in `nios/api.py `

clementtrebuchet replied:
Hello @justjais 

Ok done.

justjais commented on lib/ansible/modules/net_tools/nios/nios_ptr_record.py:
```diff
@@ -0,0 +1,159 @@
+'''
+
+EXAMPLES = '''
+- name: Create a PTR Record
```
Kindly refer the nios other module example section, as to keep nios_ptr_record module examples in sync with other modules example section. As, examples should be such which can directly be used by user to run the particular scenarios

clementtrebuchet replied:
Hello @justjais 

Ok done

justjais commented on lib/ansible/modules/net_tools/nios/nios_ptr_record.py:
```diff
@@ -0,0 +1,159 @@
+def main():
+    # Module entry point
+    ib_spec = dict(
+        name=dict(required=False),
```
Can you please refer `nios_ptr` documentation and check if user can update/modify the record ptr name section. And, if yes then please refer the `nios_host_record` module to check the implementation for updating/modifying the name field by giving name as dictionary as `{new_name:'', old_name:''}`

clementtrebuchet replied:
Hello @justjais 

Users cannot update/modify the record ptr name or ptrdname, just create or search.

![image](https://user-images.githubusercontent.com/8559910/42318712-2f2724ac-8050-11e8-9c0e-66b1575e8507.png)

![image](https://user-images.githubusercontent.com/8559910/42318782-5e6185e6-8050-11e8-8b70-3cd00cfeaa79.png)

justjais commented on test/integration/targets/nios_ptr_record/tasks/nios_ptr_record_idempotence.yml:
```diff
@@ -0,0 +1,68 @@
+    state: absent
+    provider: "{{ nios_provider }}"
+  register: ipv4_ptr_delete2
+
```
Can you please include test for ipv6 addr as well, just once test for creating Ipv6addr record will be sufficient

clementtrebuchet replied:
Hello @justjais 

Ok done.

justjais commented:
@clementtrebuchet Ok, I have reviewed the PR and have posted some comments can you please reply to respective comments. Also, I have asked @jctanner to review your changes in `flaskapp.py`. So, once completed you can go ahead and fire the build again and if all's green you can go ahead and merge the PR.

clementtrebuchet committed:
add the define nios specific constant for PTR:RECORD

clementtrebuchet committed:
fix: documentation style

clementtrebuchet committed:
add create an ipv6 ptr record

clementtrebuchet committed:
rename class name, add ipv6 unittest

clementtrebuchet commented:
Hello @justjais 

I made the changes you requested, but I assume the CI will fail again since the [PR](https://github.com/ansible/nios-test-container/pull/1) is not yet reviewed by @jctanner

clementtrebuchet commented:
Hello @justjais

It seems that the [PR](https://github.com/ansible/nios-test-container/pull/1) is ready to be merged.

justjais commented:
@clementtrebuchet yes, you can merge the PR but kindly make sure that you run the shippable tests again and everything is green before merging.

jctanner commented:
@clementtrebuchet there are test failures with this PR ...

```
2018-07-05 11:39:38 fatal: [testhost]: FAILED! => {
2018-07-05 11:39:38     "changed": false, 
2018-07-05 11:39:38     "module_stderr": "No handlers could be found for logger \"infoblox_client.utils\"\nTraceback (most recent call last):\n  File \"/tmp/ansible_UDXw9V/ansible_module_nios_ptr_record.py\", line 159, in <module>\n    main()\n  File \"/tmp/ansible_UDXw9V/ansible_module_nios_ptr_record.py\", line 153, in main\n    result = wapi.run(NIOS_PTR_RECORD, ib_spec)\n  File \"/tmp/ansible_UDXw9V/ansible_modlib.zip/ansible/module_utils/net_tools/nios/api.py\", line 238, in run\n  File \"/tmp/ansible_UDXw9V/ansible_modlib.zip/ansible/module_utils/net_tools/nios/api.py\", line 147, in _invoke_method\n  File \"/tmp/ansible_UDXw9V/ansible_modlib.zip/ansible/module_utils/net_tools/nios/api.py\", line 179, in handle_exception\nTypeError: argument of type 'NoneType' is not iterable\n", 
2018-07-05 11:39:38     "module_stdout": "", 
2018-07-05 11:39:38     "msg": "MODULE FAILURE", 
2018-07-05 11:39:38     "rc": 1
2018-07-05 11:39:38 }
```

```
2018-07-05 11:39:43 fatal: [testhost]: FAILED! => {
2018-07-05 11:39:43     "changed": false,
2018-07-05 11:39:43     "module_stderr": "Could not decode reply into json: b'<!DOCTYPE HTML PUBLIC \"-//W3C//DTD HTML 3.2 Final//EN\">\\n<title>500 Internal Server Error</title>\\n<h1>Internal Server Error</h1>\\n<p>The server encountered an internal error and was unable to complete your request.  Either the server is overloaded or there is an error in the application.</p>\\n'\nTraceback (most recent call last):\n  File \"/tmp/ansible_yw48h26t/ansible_modlib.zip/ansible/module_utils/net_tools/nios/api.py\", line 144, in _invoke_method\n  File \"/usr/local/lib/python3.6/dist-packages/infoblox_client/connector.py\", line 41, in callee\n    return func(*args, **kwargs)\n  File \"/usr/local/lib/python3.6/dist-packages/infoblox_client/connector.py\", line 364, in create_object\n    code=r.status_code)\ninfoblox_client.exceptions.InfobloxCannotCreateObject: Cannot create 'record:ptr' object(s): b'<!DOCTYPE HTML PUBLIC \"-//W3C//DTD HTML 3.2 Final//EN\">\\n<title>500 Internal Server Error</title>\\n<h1>Internal Server Error</h1>\\n<p>The server encountered an internal error and was unable to complete your request.  Either the server is overloaded or there is an error in the application.</p>\\n' [code 500]\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/tmp/ansible_yw48h26t/ansible_module_nios_ptr_record.py\", line 159, in <module>\n    main()\n  File \"/tmp/ansible_yw48h26t/ansible_module_nios_ptr_record.py\", line 153, in main\n    result = wapi.run(NIOS_PTR_RECORD, ib_spec)\n  File \"/tmp/ansible_yw48h26t/ansible_modlib.zip/ansible/module_utils/net_tools/nios/api.py\", line 238, in run\n  File \"/tmp/ansible_yw48h26t/ansible_modlib.zip/ansible/module_utils/net_tools/nios/api.py\", line 147, in _invoke_method\n  File \"/tmp/ansible_yw48h26t/ansible_modlib.zip/ansible/module_utils/net_tools/nios/api.py\", line 179, in handle_exception\nTypeError: argument of type 'NoneType' is not iterable\n",
2018-07-05 11:39:43     "module_stdout": "",
2018-07-05 11:39:43     "msg": "MODULE FAILURE",
2018-07-05 11:39:43     "rc": 1
2018-07-05 11:39:43 }
```

clementtrebuchet commented:
@justjais 
Did the container has been upgraded with the new version 
I justed pull it locally and the PR is not reflected inside, maybe you have to do something to update the container on the CI on your side ?
